### DISCUSSION TITLE
Configuring Windows Desktop access by agent mode

### DISCUSSION THREAD

**Discussion by akolahdouzan** [gravitational/teleport#9158] [Q&A]
Hello,
I want to configure windows desktop access by agent mode. There is just a description about agent mode and Gateway mode in the #RFD 34. with this consideration, there is mentioned that "Agent mode is for running a windows_desktop_service instance on the Windows host as a system service." but in the windows binary, there is only tsh client and there is nothing to run as a system service to be configured in windows host. Please help me to understand, how can I configure Remote desktop Access through agent mode. because I don't want  to use LDAPs and domain controller and another linux machine to configure as a teleport agent in middle. Thank you.

**Comment by darius-fritsch** [comment]
As far I can tell, there is a client agent in development by @zmb3 . Would suggest waiting for the upcoming update release (planned 06. Dec 2021 - Version 8.1 milestone)

Also is the desktop access feature still in a preview state (Teleport Version 8.0)

**Comment by zmb3** [comment]
Thanks for the ping, @darius-fritsch. Yes, Desktop Access is still in preview for Teleport 8. You can expect that it will gain more of the Teleport features you know and love (per-session MFA, session recording, etc) in subsequent releases.

As for @akolahdouzan's question - unfortunately agent mode hasn't been developed yet. The RFD is still an accurate depiction of our plans for that feature, and if you have feedback on how you'd like it to work we'd love to hear it!

**Comment by andreabarbaglia-criticalcase** [comment]
Hi guys, is there any news on the development of the Windows agent?
Is there any expected release date of the agent?
I am also interested in this feature that would allow a light and fast implementation of access to Windows machines without the need for AD domains.
Thank you

**Comment by ndss8989** [comment]
So it's still not possible to teleport to windows hosts, without also running a linux host running the agent, from which the windows host must allow inbound tcp/3389?

I would've thought this was a very common use case? RFD-34 sounds like it was planned? https://github.com/gravitational/teleport/blob/master/rfd/0034-desktop-access-windows.md

(actually, ideally, both the auth and the agent could easily be installed in one easy way; our use case is as an MSP needing to teleport to customer hosts both onprem and cloud and ideally onboarding them is easy as possible; requiring minimal download and config)

**Reply by webvictim** [reply]
> So it's still not possible to teleport to windows hosts, without also running a linux host running the agent, from which the windows host must allow inbound tcp/3389?

Correct.

> I would've thought this was a very common use case? RFD-34 sounds like it was planned? https://github.com/gravitational/teleport/blob/master/rfd/0034-desktop-access-windows.md

It's come up occasionally, but building Teleport for Windows is non-trivial and would require considerable engineering effort.

> (actually, ideally, both the auth and the agent could easily be installed in one easy way; our use case is as an MSP needing to teleport to customer hosts both onprem and cloud and ideally onboarding them is easy as possible; requiring minimal download and config)

You can run the Teleport `auth_service`, `proxy_service` and `windows_desktop_service` on the same machine if needed with a few simple changes to the configuration file.
